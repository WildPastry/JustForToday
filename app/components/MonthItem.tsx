import { ColorSchemeName, Pressable, StyleSheet } from 'react-native';
import { IDate, IMonthItem } from '../types/date.types';
import { AppState } from '../redux/store';
import Colours from '../constants/Colours';
import { Text } from './styles/Themed';
import { useAppSelector } from '../redux/hooks';
import useColorScheme from '../hooks/useColorScheme';

const MonthItem: React.FC<IMonthItem> = (props: IMonthItem): JSX.Element => {
  // Component settings
  const colorScheme: NonNullable<ColorSchemeName> = useColorScheme();

  // Data from store
  const dates: IDate = useAppSelector((state: AppState): IDate => {
    return state.date;
  });

  const isCurrentMonth = (): boolean => {
    return dates.currentMonth === props.id;
  };

  return (
    <Pressable
      style={[
        styles.monthItem,
        isCurrentMonth()
          ? { backgroundColor: Colours[colorScheme].currentBtn }
          : { backgroundColor: Colours[colorScheme].btn }
      ]}
      onPress={props.onPress}>
      <Text
        style={[
          styles.text,
          isCurrentMonth()
            ? { color: Colours[colorScheme].currentBtnText }
            : null
        ]}>
        {props.id}
      </Text>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  text: {
    fontSize: 15,
    lineHeight: 21,
    textAlign: 'center'
  },
  monthItem: {
    borderRadius: 12,
    marginBottom: 20,
    paddingVertical: 20
  }
});

export default MonthItem;
