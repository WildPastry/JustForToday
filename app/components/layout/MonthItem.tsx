import { ColorSchemeName, Pressable, StyleSheet } from 'react-native';
import { IDate, IMonthItem } from '../../types/date.types';
import { AppState } from '../../redux/store';
import { Text } from '../styles/Themed';
import { useAppSelector } from '../../redux/hooks';
import useColorScheme from '../../hooks/useColorScheme';

const MonthItem: React.FC<IMonthItem> = (props: IMonthItem): JSX.Element => {
  // Component settings
  const colorScheme: NonNullable<ColorSchemeName> = useColorScheme();
  const currentMonthTheme = styles[`${colorScheme}CurrentMonth`];

  // Data from store
  const dates: IDate = useAppSelector((state: AppState): IDate => {
    return state.date;
  });

  const isCurrentMonth = (): boolean => {
    return dates.currentMonth === props.id;
  };

  return (
    <Pressable
      style={[
        styles.monthItem,
        isCurrentMonth() ? currentMonthTheme : styles[`${colorScheme}MonthItem`]
      ]}
      onPress={props.onPress}>
      {/* Month item */}
      <Text style={styles.text}>{props.id}</Text>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  text: {
    textAlign: 'center'
  },
  monthItem: {
    borderRadius: 12,
    marginBottom: 12,
    paddingVertical: 12
  },
  lightMonthItem: {
    backgroundColor: '#e5edf9'
  },
  darkMonthItem: {
    backgroundColor: '#171b43'
  },
  lightCurrentMonth: {
    borderColor: '#131324',
    borderWidth: 0.5
  },
  darkCurrentMonth: {
    backgroundColor: '#131324'
  }
});

export default MonthItem;
