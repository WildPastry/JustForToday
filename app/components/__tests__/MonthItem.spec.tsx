import MonthItem from '../MonthItem';
import renderer from 'react-test-renderer';

describe('<MonthItem />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MonthItem id={''} name={''} days={[]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
