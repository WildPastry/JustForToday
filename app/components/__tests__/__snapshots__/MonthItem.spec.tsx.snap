// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonthItem /> renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "borderRadius": 12,
        "marginBottom": 20,
        "paddingVertical": 20,
      },
      {
        "backgroundColor": "#e5edf9",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        [
          {
            "textAlign": "center",
          },
          null,
        ],
      ]
    }
  />
</View>
`;
